packages/@react-aria/utils/test/mergeProps.test.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import classNames from 'classnames';
+import {mergeIds} from '../src/useId';
+import {mergeProps} from '../';
+
+
+describe('mergeProps', function () {
+  it('handles one argument', function () {
+    let onClick = () => {};
+    let classname = 'primary';
+    let id = 'test_id';
+    let mergedProps = mergeProps({onClick, classname, id});
+    expect(mergedProps.onClick).toBe(onClick);
+    expect(mergedProps.classname).toBe(classname);
+    expect(mergedProps.id).toBe(id);
+  });
+
+  it('combines callbacks', function () {
+    console.log = jest.fn();
+    let message1 = 'click1';
+    let message2 = 'click2';
+    let message3 = 'click3';
+
+    let a = {onClick: () => console.log(message1)};
+    let b = {onClick: () => console.log(message2)};
+    let c = {onClick: () => console.log(message3)};
+    let mergedProps = mergeProps(a, b, c);
+    mergedProps.onClick();
+    expect(console.log).toHaveBeenCalledWith(message1);
```

Comment:
Would you mind verifying the order they are called in too? We don't want to accidentally change that on people

Reply:
Good idea

Reply:
@snowystinger changed to `toHaveBeenNthCalledWith`

Reply:
Thanks!